mlflow/onnx.py

```diff
@@ -170,10 +170,17 @@ def predict(self, dataframe):
             }
         else:
             feed_dict = {self.inputs[0][0]: dataframe.values}
-
         predicted = self.rt.run(self.output_names, feed_dict)
-        return pd.DataFrame.from_dict(
-            {c: p.reshape(-1) for (c, p) in zip(self.output_names, predicted)})
+
+        def format_output(data):
+            # Output can be list and it should be converted to a numpy array
+            # https://github.com/mlflow/mlflow/issues/2499
+            if isinstance(data, list):
+                data = np.asarray(data)
+            return data
+        response = pd.DataFrame.from_dict({c: format_output(p)
+                                           for (c, p) in zip(self.output_names, predicted)})
```

Comment:
@andychow-db Looks like we're no longer reshaping the numpy array output. This is necessary in order to flatten multi-dimensional arrays into Pandas DataFrame columns. I would expect pyfunc tests to fail without this reshaping logic.

Reply:
Nice catch! Readded the reshape(-1) function.